Firestore/core/test/firebase/firestore/immutable/sorted_set_test.cc

```diff
@@ -94,21 +94,6 @@ TEST(SortedSetTest, IteratorsAreDefaultConstructible) {
       "is default constructible");
 }
 
-TEST(SortedSetTest, CanBeConstructedFromSortedMap) {
-  using Map = SortedMap<int, int>;
```

Comment:
What's the reason to remove this?

Reply:
This feature of being able to construct a `SortedSet` from an arbitrary `SortedMap` required `SortedSet` to have this declaration:

```c++
template <typename K,
          typename C = util::Comparator<K>,
          typename V = util::Empty,
          typename M = SortedMap<K, V, C>>
class SortedSet;
```

And that meant that forward declaring `DocumentKeySet` required all of this:

```c++
namespace util {
struct Empty;

template <typename T>
class Comparator;
}  // namespace util

namespace immutable {
template <typename K, typename V, typename C>
class SortedMap;

template <typename K, typename C, typename V, typename M>
class SortedSet;
}  // namespace immutable

namespace model {
class DocumentKey;

using DocumentKeySet =
    SortedSet<DocumentKey,
              util::Comparator<DocumentKey>,
              util::Empty,
              SortedMap<DocumentKey, util::Empty, util::Comparator<DocumentKey>>>;
}  // namespace model
```

With this simpler signature, forward declarations of `SortedSet` types become much simpler because we no longer have to mention `Empty` or `SortedMap`:

```c++
namespace util {
template <typename T>
class Comparator;
}  // namespace util

namespace immutable {
template <typename K, typename C>
class SortedSet;
}  // namespace immutable

namespace model {
class DocumentKey;

using DocumentKeySet = immutable::SortedSet<DocumentKey, util::Comparator<DocumentKey>>;
}  // namespace model
```

(Note that forward declarations can't have default template arguments because C++ doesn't let you redeclare template defaults. Were to do declare default parameters here we couldn't have defaults on the actual declaration.)